warehouse/authnz/_permissions.py

```diff
@@ -0,0 +1,41 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import StrEnum
+
+
+class Permissions(StrEnum):
+    """
+    Permissions can be specified in an ACL (`__acl__`) or `@view_config(permission=...)`
+    instead of using a string literal, minimizing the chance of typos.
+
+    They are also disconnected from Principals (users, groups, etc.), so they can be
+    used in a more generic way. For example, a permission could be used to allow a user
+    to edit their own profile, or to allow a group to edit any profile.
+
+    Naming should follow the format:
+        <optional scope>:<resource>:<action>
```

Comment:
After writing this, I'm realizing that after all is said and done, the actual strings themselves don't matter, as long as they are consistently used across the codebase.

If that holds true, then we can further simplify by using the StrEnum feature `= auto()` and not have to worry what the string is. We could likely change the format to `ScopeResourceAction = auto()`.